src/ngx_rewrite_driver_factory.cc

```diff
@@ -151,6 +151,10 @@ RewriteOptions* NgxRewriteDriverFactory::NewRewriteOptions() {
   return options;
 }
 
+RewriteOptions* NgxRewriteDriverFactory::NewRewriteOptionsForQuery() {
+  return new NgxRewriteOptions(thread_system());
```

Comment:
I see.  In Apache, the kCoreFilters setting is in create_dir_config, not in ApacheRewriteDriverFactory::NewRewriteOptions.  This is still different from what you have, where NgxRewriteDriverFactory::NewRewriteOptions *does* call config->SetDefaultRewriteLevel(RewriteOptions::kCoreFilters);

I wonder if there are still paths where these differences will bite us.